Create 2025-04-09.md (#726)
solid/specification: meetings/2025-04-09.md

* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Ethics and Professional Conduct](https://www.w3.org/Consortium/cepc/)
* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
* If this is your first time, welcome! please introduce yourself.

### Scribes

* elf Pavlik

## Topics

### Peaceful Progress
(topic moved forward from [21 may 2025 agenda](https://hackmd.io/dtohRN8ZSVqBgRElSwEZlw))

* MdJ: In the interest of our community's social harmony, let's focus our work as a CG on topics that don't overlap with topics that other groups within the Solid Community may have principled opinions about.
  ... And it's not just about the social harmony. There is also a technical requirement. The power of Solid is that we're creating a _standard_ way for folks to run and use their personal online data stores. Creating a non-standard way to do that would be trivial, but creating a standard way requires convergence on many levels (client-server as well as client-client).

>   ... How can we achieve this convergence which is so essential for Solid? Would it for instance be better for the Solid Project as a whole if for instance Solid CG and Solid Lab Flanders would coordinate more? Or if Solid OS would coordinate more with SAI? Otherwise I fear we can continue to incubate many parallel experiments for a long time, without delivering on Solid's promise.; [JZ: and relative scope of CG compared to Practitioners]

Reviewer: ```suggestion
  ... How can we achieve this convergence which is so essential for Solid? Would it be better for the Solid Project as a whole if, for instance, Solid CG and Solid Lab Flanders would coordinate more? Or if Solid OS would coordinate more with SAI? Otherwise I fear we will continue to incubate many parallel experiments for a long time, without delivering on Solid's promise. [JZ: and relative scope of CG compared to Practitioners]
```

> * eP: CGs are a bit informal, we cannot expect too much from the reports they produce. They are incubation spaces and can explore different directions. The expectaction from the LWS WG  which has more formal process is to publish specifications. So as a CG we shouldn't make too far-fetched promises about stability. We only incubate, there is zero stability. We all can support the WG to keep moving forward and to resolve objections using the established process, and that can maybe help us avoid similar situations (like what happened with PR 711) in the future.

Reviewer: ```suggestion
* eP: CGs are a bit informal; we cannot expect too much from the reports they produce. They are incubation spaces and can explore different directions. The expectaction from the LWS WG, which has more formal process, is to publish specifications. So, as a CG, we shouldn't make too far-fetched promises about stability. We only incubate; there is zero stability. We all can support the WG to keep moving forward and resolve objections using the established process, and that can maybe help us avoid situations like what happened with [PR 711](https://github.com/solid/specification/pull/711) in the future.
```

> * JZ: In relation to WG, we will always be ahead of them. We are trying things out and we don't want to do anything that WG decided. Isn't it CG's job to experiment? This can serve as input for the WG decisions. If they decide differently we will have to reevaluate that.

Reviewer: ```suggestion
* JZ: In relation to WG, we will always be ahead of them. We are trying things out, and we don't want to do anything that WG decided. Isn't it CG's job to experiment? This can serve as input for the WG decisions. If they decide differently, we will have to reevaluate that.
```

* MdJ: I don't know whether it is it even worth incubating things like patch. Would it help more to focus on client-client topics?

> * JZ: If we want to get adoption, we need a product that functiions in a way that people can use. It does seem to be our role to incubate proposals serving that.

Reviewer: ```suggestion
* JZ: If we want to get adoption, we need a product that functions in a way that people can use. It does seem to be our role to incubate proposals serving that.
```

* MdJ: The patch PR came from SolidOS issues with CSS. 

> * eP: Do we agree that the CG's work does not give stability, it is all subject to change, and if you adopt early, you participate in the incubation. These are just CG drafts, we should communicate that clearly.

Reviewer: ```suggestion
* eP: Do we agree that the CG's work does not give stability, it is all subject to change, and if you adopt early, you participate in the incubation? These are just CG drafts; we should communicate that clearly.
```

* HZ: I want to point out that this is a dangerous message to send since it can hurt adoption.

> * JZ: I will bridge what Pavlik and Hadrian say. Wherever we diverge from WG we should make it explicit. This way people will know 
> * eP: I'm not sure I understand, the WG doesn't have a FPWD, it only has use cases and requirements. The CG should be tracking the WG, and refer to what they publish, when they publish it.

Reviewer: ```suggestion
* JZ: I will bridge what Pavlik and Hadrian say. Wherever we diverge from WG, we should make it explicit. This way people will know.
* eP: I'm not sure I understand. The WG doesn't have a FPWD, it only has use cases and requirements. The CG should be tracking the WG, and refer to what they publish, when they publish it.
```

> * MdJ: I also thought about that, one one side we say we are incubating and open for experiments, but what is the effect on perception of Solid. Will in few years one protocol will be selected from it. Will people just say they will wait.
> * JZ: Was there a feedback from WG that they object direction we are taking?

Reviewer: ```suggestion
* MdJ: I also thought about that. On one side, we say we are incubating and open for experiments, but what is the effect on perception of Solid? In a few years, one protocol will be selected from it. Will people just say they will wait?
* JZ: Was there a feedback from the WG that they object to a direction we are taking?
```

> * MdJ: Aaron sent email giving ok to make changes as long as we don't publish 0.12. When we started doing it the resistance was more from SolidLab than the WG. We decided something in CG, published to solidcommunity.net and it looked that this is de-facto the standard. 

Reviewer: ```suggestion
* MdJ: Aaron sent email giving OK to make changes, as long as we don't publish 0.12. When we started doing it, the resistance was more from SolidLab than the WG. We decided something in CG, published it to solidcommunity.net and it looked like this was de-facto the standard. 
```

* JZ: Is there no way to convince them to participate in CG?

> * eP: you say that CG decisions through pivot to solidcommunity.net become de-facto standards, but I don't see it that way, and it's misinforming people. What the CG does have zero standing until the WG adopts it. We can say we avoid making changes without technical commitment, but the most honest way is to say our work has no formal standing.
> * MdJ: I don't think we will reach conclusion or solution. Probably we can reach PR 711 and be aware of the dynamics.

Reviewer: ```suggestion
* eP: You say that CG decisions through pivot to solidcommunity.net become de-facto standards, but I don't see it that way, and it's misinforming people. What the CG does has zero standing until/unless the WG adopts it. We can say we avoid making changes without technical commitment, but the most honest way is to say our work has no formal standing.
* MdJ: I don't think we will reach conclusion or solution. Probably we can reach [PR 711](https://github.com/solid/specification/pull/711) and be aware of the dynamics.
```

* T: ... unstable solidcommunity.net
* JZ: There is already testing insteance.

> * MdJ: What does it mean that something gets merged in the spec repo. Someone runs into an issue using solidcommunity.net. Makes a PR to the specification repo, it gets merged, deployed to solidcommunity.net. If everyone runs their own server coherence gets even lower. Solidcommunity.net is used by SolidOS team to do demos etc. If we don't update that server we may don't bother updating the specification repo. 

Reviewer: ```suggestion
* MdJ: What does it mean that something gets merged in the spec repo? Someone runs into an issue using solidcommunity.net, makes a PR to the specification repo. It gets merged, deployed to solidcommunity.net. If everyone runs their own server, coherence gets even lower. Solidcommunity.net is used by SolidOS team to do demos, etc. If we don't update that server, we may not bother updating the specification repo. 
```

> * eP: Even though the repo is called 'specification', we could include some more inline notes in cases where there is no full concensus.
> * MdJ: This way we can't have a test suite. We can't have compatible implementations. We already have some servers doing WAC, some ACP, some SAI others do UMA. 
> * MdJ: If Solid is a way to signal what can be done but there is no consistent way of achieving it. ...

Reviewer: ```suggestion
* eP: Even though the repo is called 'specification', we could include some more inline notes where there is no full consensus.
* MdJ: This way we can't have a test suite. We can't have compatible implementations. We already have some servers doing WAC, some ACP, some SAI, others do UMA. 
* MdJ: If Solid is a way to signal what can be done, but there is no consistent way of achieving it ...
```

> * JW: The CG is here to incubate specs. It should be a goal of consensus rather than specification divergance. For issue 711, that was particularly controversial, I don't think we will hit that many issues with this degree of controversy. 

Reviewer: ```suggestion
* JW: The CG is here to incubate specs. It should be a goal of consensus rather than specification divergence. For [PR 711](https://github.com/solid/specification/pull/711), that was particularly controversial. I don't think we will hit many issues with this degree of controversy. 
```

* MdJ: WAC vs. ACP, SAI, UMA.

> * JW: for those, my view that CG should be incubating both WAC and ACP. We can reach consensus and LWS WG can look at them and decide what will be put forward in LWS spec and have more contancious debate there. For SAI vs. Type indexes etc. Specs that are not in scope of LWS, we need to decide how to choose from them. This should be done in data driven maner. Seeing how the cover the functional requirements. Data interop spec should be one 

Reviewer: ```suggestion
* JW: For those, my view is that CG should be incubating both WAC and ACP. We can reach consensus, and LWS WG can look at them and decide what will be put forward in LWS spec and have more contentious debate there. For SAI vs. Type indexes, etc. We need to decide how to choose from Specs that are not in scope of LWS. This should be done in a data-driven manner. Seeing how they cover the functional requirements. Data interop spec should be one.
```

> * eP: For example with SAI it's a little bit more complex, it's not SAI vs type indexes, there is also quite a lot about authorization there, and for the LWS WG there are VCs and data grants in scope, so there is some overlap there.
> * eP: I'm happy that JW refers to requirements, like the WG, the CG needs to work with scopes and requirements. The conversations often seem to be emotional and sentimental. But we should look at which requirements each proposal satisfies.

Reviewer: ```suggestion
* eP: For example, with SAI, it's a little bit more complex; it's not SAI vs type indexes, there is also quite a lot about authorization there, and for the LWS WG, VCs and data grants are in scope, so there is some overlap there.
* eP: I'm happy that JW refers to requirements. Like the WG, the CG needs to work with scopes and requirements. The conversations often seem to be emotional and sentimental. But we should look at which requirements each proposal satisfies.
```

> * JW: The data grants, I belive are mentikoned in LWS charter. In ESS consent flow VCs are used as part of access requrest mechanism. That has been put into ACP via `acp:vc`. I would have a conversation with Aaron as well as Wouter who is working on UMA flow for CSS. You could see if you can come to rought consensus. Otherwise this can split into different specs.

Reviewer: ```suggestion
* JW: I believe data grants are mentioned in LWS charter. In ESS, consent flow VCs are used as part of the access request mechanism. That has been put into ACP via `acp:vc`. I would have a conversation with Aaron as well as Wouter, who is working on UMA flow for CSS. You could see if you can come to rough consensus. Otherwise, this can split into different specs.
```

* eP: ...
* JW: I put some links in the chat about other implementations of Inrupt specific flow
  * https://amc.inrupt.com/login?returnTo=%2F
  * https://github.com/openwallet-foundation-labs/solid-data-wallet

> * JW: For 711 specifically it would be good to merge it.

Reviewer: ```suggestion
* JW: For [PR 711](https://github.com/solid/specification/pull/711) specifically, it would be good to merge it.
```

> * MdJ: It seems that it cooled down a bit, we can move forward.
> * MdJ: I think we had some conversation of what it means to call our repos specifications. While we are incubating we should be aware of the rest of the Solid project.
> * MdJ: In the future when we make a spec change, we can add a note worning that WG can change it down the road.

Reviewer: ```suggestion
* MdJ: It seems that it cooled down a bit. We can move forward.
* MdJ: I think we had some conversation of what it means to call our repo `specification`. While we are incubating, we should be aware of the rest of the Solid project.
* MdJ: In the future, when we make a spec change, we can add a note warning that a WG can change it down the road.
```

> * JZ: So far there has been a very good relation between Practicioners and CG. When we have outreach, we should either do it with Practicioners or joint meeting. For example catalogue, one part would be community focued and another part like shapes fits more CG. We should work on that jointly instead of discussing outreach issues just here in the CG. For example having an Awesome list would be more of Practicioners topic. We will do our part in keeping CG involved on how we are progressing. I would appreciate if CG keeps Practiciones involved when it comes to outreach and community involvement.

Reviewer: ```suggestion
* JZ: So far there has been a very good relation between Practitioners and CG. When we have outreach, we should either do it with Practitioners or joint meeting. For example, catalogue, one part would be community focused, and another part, like shapes, fits more CG. We should work on that jointly instead of discussing outreach issues just here in the CG. For example, having an Awesome list would be more of a Practitioners topic. We will do our part in keeping CG involved on how we are progressing. I would appreciate if CG keeps Practitioners involved when it comes to outreach and community involvement.
```

### FedCM 

> * Th: FedCM is on its way, but the IdP registration feature ( which could strongly improve Solid UX ) is still not part of core. I would like to share the current situation, and gather advice and feedback ( from people you have experience with the W3C process ) on how we can build a stronger case to make this feature part of FedCM core. 

Reviewer: ```suggestion
* Th: FedCM is on its way, but the IdP registration feature (which could strongly improve Solid UX) is still not part of core. I would like to share the current situation, and gather advice and feedback (from people who have experience with the W3C process) on how we can build a stronger case to make this feature part of FedCM core. 
```

> * Th: I'm not used to W3C. Idp registration feature, which is needed for Solid is not part of FedCM and still working on. I don't know how to proceed with google developers which are leading the spec.I don't want to implement it if there is no usage. I don't know if it is only Google's thing to be in that position to listen of what peole want. This feature doesn't have direct interest from google since they have strong position in IdP world. I'm bit lost in all that. 

Reviewer: ```suggestion
* Th: I'm not used to W3C. IdP registration feature, which is needed for Solid, is not part of FedCM and still being worked on. I don't know how to proceed with Google developers who are leading the spec. I don't want to implement it if there is no usage. I don't know if it is only Google's thing to be in that position to listen to what people want. This feature doesn't have direct interest from Google since they have a strong position in IdP world. I'm a bit lost in all that. 
```

> * eP: Thanks for bringing it up. I'll try to attend the all-day hybrid Fed-id CG meeting on Friday. I've seen your demo. Do you think it's ready to deploy it and ask other apps to also adopt it, so we have some implementation? Some signalling/endorsement from LWS WG would also be nice. Jesse also wanted to integrate with the Fediverse, maybe that could be another possibility. And I still need to understand the 'lightweight' FedCM. My understanding was that it also has IDP Registration functionality? I'll have a call with NLNet on Friday. I could help implement the lightweight one, maybe.

Reviewer: ```suggestion
* eP: Thanks for bringing it up. I'll try to attend the all-day hybrid FedID CG meeting on Friday. I've seen your demo. Do you think it's ready to deploy and ask other apps to also adopt it, so we have some implementation? Some signalling/endorsement from LWS WG would also be nice. Jesse also wanted to integrate with the Fediverse; maybe that could be another possibility. And I still need to understand the 'lightweight' FedCM. My understanding was that it also has IdP Registration functionality? I'll have a call with NLNet on Friday. I could help implement the lightweight one, maybe.
```

> * Th: IdP Registration could benefit from the ecosystem. In univerisities there are a lot of systems that can benefit from the feature. If you want to coordinate and make people aware of that.
> * Th: FedCM can be implemented in production. You need to change cookie policies which would make solidcommunity insecure. 
> * MdJ: Did you talk about that with Aaron Parecki and Sam Goto?

Reviewer: ```suggestion
* Th: IdP Registration could benefit from the ecosystem. In universities, there are a lot of systems that can benefit from the feature. If you want to coordinate and make people aware of that.
* Th: FedCM can be implemented in production. You need to change cookie policies which could make solidcommunity insecure.
* MdJ: Did you talk about that with Aaron Parecki and Sam Goto?
```

> * Th: Someone found a workaround with two cookie. This would require a lot of work in CSS.
> * MdJ: Are they aware of your work? When I raise the issue they just point to that workaround with a second cookie. It is only for IdP Registration which is not part of the core. https://github.com/w3c-fedid/FedCM/issues/587

Reviewer: ```suggestion
* Th: Someone found a workaround with two cookies. This would require a lot of work in CSS.
* MdJ: Are they aware of your work? When I raise the issue, they just point to that workaround with a second cookie. It is only for IdP Registration, which is not part of the core. https://github.com/w3c-fedid/FedCM/issues/587
```

### Catalog Awesome List

* JZ: Brief update on discussion Alain and I had about his Awesome List suggestion.  See https://github.com/solid/catalog/issues/6#issuecomment-2787659110

> * JZ: This goes to my point about coordination between CG and Practitioners. As an example of link shaerd in next topic. List of those extensions is useful to people who start servers or who are creating extensions. Catalogue listing can list which servers use which extensions — this is useful primarily to users deciding on which server to select. Another approach is if servers have discovery for which extensions they are supporting - this is useful for developers to know how the app should behave on each server. We need social context not only listing of technical features. If we want to make it useful to the whole community we may need a wider view. Shape for server extension needs feedback (see link above). I hope it is done in a way that it takes into account other types of users. 

Reviewer: ```suggestion
* JZ: This goes to my point about coordination between CG and Practitioners. As an example of link shaerd in next topic. List of those extensions is useful to people who start servers or who are creating extensions. Catalogue listing can list which servers use which extensions. This is useful primarily to users deciding on which server to select. Another approach is making servers have discovery for which extensions they support. This is useful for developers to know how the app should behave on each server. We need social context, not only listing of technical features. If we want to make it useful to the whole community, we may need a wider view. Shape for server extension needs feedback (see link above). I hope it is done in a way that takes other types of users into account.
```

### Curated list of known CSS extensions

https://github.com/CommunitySolidServer/CommunitySolidServer/discussions/2024

* eP: Follow up on my action from last week

### Delete backup data from solidcommunity.net before migration to PIVOT

> * MdJ: Is that just an announcement of we should move it to next week.
> * AB: The question was made by Jesse. Before deleting it I want to make sure everyone is in agreement.
> * JW: Unless objections I would like those volumes to be deleted.
> * MdJ: Normally we have volumes with backup. Can we keep them for a couple of months and delete them let's say 1st of July. In case some bugs are still found

Reviewer: ```suggestion
* MdJ: Is that just an announcement or should we move it to next week?
* AB: The question was made by Jesse. Before deleting it, I want to make sure everyone is in agreement.
* JW: Barring objections, I would like those volumes to be deleted.
* MdJ: Normally, we have volumes with backup. Can we keep them for a couple of months and delete them, let's say, July 1? In case some bugs are still found.
```
